client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/tcp/RoundRobinBalancingStrategy.java

```diff
@@ -28,7 +29,7 @@ public void setServers(Collection<SocketAddress> servers) {
       this.servers = servers.toArray(new SocketAddress[servers.size()]);
       // keep the old index if possible so that we don't produce more requests for the first server
       if (index >= this.servers.length) {
-         index = 0;
+         index = ThreadLocalRandom.current().nextInt(this.servers.length);
```

Comment:
Shouldn't `index` be initialized to MAX_INT in order to trigger this on 1st call ?

Reply:
Oops, I meant to remove the `if`